src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -329,7 +330,8 @@ internal IEnumerable<DiagnosticAnalyzer> GetAnalyzers(Project project)
                 var versions = new VersionArgument(projectTextVersion, semanticVersion, projectVersion);
                 foreach (var stateSet in _stateManager.GetOrUpdateStateSets(project))
                 {
-                    if (SkipRunningAnalyzer(project.CompilationOptions, analyzerDriver, openedDocument: false, skipClosedFileChecks: false, stateSet: stateSet))
+                    // Compilation actions can report diagnostics on open files, so we skipClosedFileChecks.
+                    if (SkipRunningAnalyzer(project.CompilationOptions, analyzerDriver, openedDocument: true, skipClosedFileChecks: true, stateSet: stateSet))
                     {
```

Comment:
asking because I am not sure whether the option will be understood as perf related option or user experience related option. since the option make us to behavior like vs2013.